### DISCUSSION TITLE
Missing key(s) in state_dict: "model.33.anchor_grid"

### DISCUSSION THREAD

**Discussion by Argentum471** [zhiqwang/yolov5-rt-stack#233] [Q&A] [answered]
Hello @zhiqwang
I got this error when trying to load my custom yolov5n6 model.  Is there a way to load yolov5-v6.0 models?

`model = YOLOv5.load_from_yolov5(model_path, score_thresh=0.25, version="r6.0")`

**Comment by zhiqwang** [comment]
Hi @123556666 , Sorry for missing this ticket. Could you check the version of `yolort` you're using?

In version 0.5.2, we've supported loading custom `yolov5n6` model and auto check whether it is P6 or not.

https://github.com/zhiqwang/yolov5-rt-stack/blob/4cba0437389e2cccbcc299ab196c922945c93d45/yolort/utils/update_module_state.py#L70-L72

If it still reports an unsupported error, you can send me your model file if it's possible and I'll reproduce this bug and figure out how to fix it.

**Reply by Argentum471** [reply]
Sorry for long reply.
I am using version 0.5.2. I got model.33.anchor_grid error with loading my custom yolov5n6 model a bit earlier at
checkpoint_yolov5 = load_yolov5_model(checkpoint_path).
Pretrained yolov5n6  model loading, working and exporting fine. The error happens because of an additional anchor_grid attribute in Detect layer after training.  

![image](https://user-images.githubusercontent.com/26543304/143029917-c4bbce04-d692-4915-b6f5-7be689100028.png)

There was also a similar problem with custom yolov5 ckpt. I am using ultralytics/yolov5 v6.0 for model training.

**Reply by zhiqwang** [reply]
Seems to be caused by incompatibility with the latest version of YOLOv5 from the error log information. I'll try to fix this bug in the next two days. And I received the model checkpoint you sent, thank you again.

EDITED: I create an issue #236 for this problem, let's move the discussion there.

**Comment by zhiqwang** [comment, marked as answer]
Hi @123556666

We update the upstream YOLOv5 in #235 , please pull the main branch of `yolort` and check whether this modification will fix your problem.

**Reply by eelgaev** [reply]
I encounter the same issue.
`issing key(s) in state_dict: "model.33.anchor_grid"` 
when loading a yolov5n6 trained model with the latest ultralytics/yolov5 commit.

**Reply by zhiqwang** [reply]
Hi @Okoyl , could you try to reinstall `yolort` from source like the following
```shell
pip uninstall yolort
pip install 'git+https://github.com/zhiqwang/yolov5-rt-stack.git'
```
we plan to distribute 0.6.0 at the end of the month.
